gadgets/mri_core/GenericReconBase.cpp

```diff
@@ -22,6 +23,18 @@ namespace Gadgetron {
         {
             Gadgetron::get_debug_folder_path(debug_folder.value(), debug_folder_full_path_);
             GDEBUG_CONDITION_STREAM(verbose.value(), "Debug folder is " << debug_folder_full_path_);
+
+            // Create debug folder if necessary
+            boost::filesystem::path boost_folder_path(debug_folder_full_path_);
+            try
```

Comment:
Here we may need to first check debug folder does not exist. If so, then create it.